From 9a5301530029362ee6cc9c463e1934cff11e9b61 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:50:13 -0400
Subject: [fs] nfsd: set timeparms.to_maxval in setup_callback_client

Message-id: <1403560314-32525-72-git-send-email-bfields@redhat.com>
Patchwork-id: 84584
O-Subject: [Patch RHEL7 071/172] nfsd: set timeparms.to_maxval in setup_callback_client
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Jeff Layton <jlayton@redhat.com>

...otherwise the logic in the timeout handling doesn't work correctly.

Spotted-by: Trond Myklebust <trond.myklebust@primarydata.com>
Cc: stable@vger.kernel.org
Signed-off-by: Jeff Layton <jlayton@redhat.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 3758cf7e14b753838fe754ede3862af10b35fdac
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4callback.c b/fs/nfsd/nfs4callback.c
index 39c8ef8..2c73cae 100644
--- a/fs/nfsd/nfs4callback.c
+++ b/fs/nfsd/nfs4callback.c
@@ -654,9 +654,11 @@ static struct rpc_clnt *create_backchannel_client(struct rpc_create_args *args)
 
 static int setup_callback_client(struct nfs4_client *clp, struct nfs4_cb_conn *conn, struct nfsd4_session *ses)
 {
+ int maxtime = max_cb_time(clp->net);
  struct rpc_timeout timeparms = {
-  .to_initval = max_cb_time(clp->net),
+  .to_initval = maxtime,
   .to_retries = 0,
+  .to_maxval = maxtime,
  };
  struct rpc_create_args args = {
   .net  = clp->net,
-- 
1.7.1